### PR TITLE
Add vegan and vegetarian spin commands

### PR DESCRIPTION
**Pull request by jotjern**

### CODE REVIEW

File: config/pizza.json [hunk 1]
```diff
@@ -2,376 +2,498 @@
...
+    "name": "101 DEN SINGLE",
+    "vegetarian": true,
+    "vegan": false
   },
   {
     "description": "Ost, tomatsaus, pepperoni, r\u00f8dl\u00f8k, r\u00f8d paprika, gul paprika og oregano.",
     "extra": "Allergener: hvete, melk, selleri",
-    "name": "102 VINNEREN"
+    "name": "102 VINNEREN",
+    "vegetarian": true,
```
Comment (henrikhorluck) [inline config/pizza.json]: Den er ikke vegetar (pepperoni)

File: config/pizza.json [hunk 2]
```diff
@@ -2,376 +2,498 @@
...
-    "name": "Delt pizza"
+    "name": "51 T\u00d8FFINGEN",
+    "vegetarian": false,
+    "vegan": false
   },
   {
     "description": "Ost, tomatsaus - og bare det!",
     "extra": "Allergener: hvete, melk, selleri",
-    "name": "01 DEN ENKLE"
+    "name": "01 DEN ENKLE",
```
Comment (henrikhorluck) [inline config/pizza.json]: Vi har to av den enkle?

**Review by henrikhorluck — CHANGES_REQUESTED** [review]
Kunne muligens bare satt default på vegetar og vegansk til false?

**Comment by jotjern** [comment]
> Kunne muligens bare satt default på vegetar og vegansk til false?

🤷‍♂️

File: src/slack.rs [hunk 3]
```diff
@@ -147,9 +147,23 @@ async fn handle_disconnect_message(
...
     message: incoming::Incoming<incoming::SlashCommandIncomingMessage>,
 ) -> outgoing::SlackOutgoingMessage {
-    match message.payload.command.as_str() {
-        "/spin" => {
-            let pizza = get_random_pizza();
+    let command_spin_mode = match message.payload.command.as_str() {
+        "/spin" => Some(SpinMode::Any),
+        "/spin-vegan" => Some(SpinMode::Vegan),
+        "/spin-vegetarian" => Some(SpinMode::Vegetarian),
+        _ => None,
```
Comment (junlarsen) [inline src/slack.rs]: I would much rather early exit here, than return None and handle the case below. That way you can have spin_command_mode: SpinMode

**Review by junlarsen — CHANGES_REQUESTED** [review]
Like it, just a small code nit

**Comment by junlarsen** [comment]
Also when did we rename the repo..?

**Comment by jotjern** [comment]
> Also when did we rename the repo..?

git clone git@github.com:dotkom/pizzapicker.git
fatal: destination path 'pizzapicker' already exists and is not an empty directory.

**Review by junlarsen — APPROVED** [review]

File: src/slack.rs [hunk 4]
```diff
@@ -147,42 +147,44 @@ async fn handle_disconnect_message(
...
+        "/spin-vegan" => SpinMode::Vegan,
+        "/spin-vegetarian" => SpinMode::Vegetarian,
         _ => {
             tracing::warn!("Received unknown command: {}", message.payload.command);
-            outgoing::SlackOutgoingMessage::Empty(outgoing::Outgoing::new(
+            return outgoing::SlackOutgoingMessage::Empty(outgoing::Outgoing::new(
                 message.envelope_id,
                 None,
-            ))
+            ));
```
Comment (brage-andreas) [inline src/slack.rs:159]: Why switch to `return value;` here?

Comment (jotjern) [reply src/slack.rs:159]: Early return, the diff is hard to read

**Pull request merged**